src/psyclone/parse/file_info.py

```diff
@@ -499,58 +465,48 @@ def get_fparser_tree(
         # We directly call the cache saving routine here in case that the
         # fparser tree will be modified later on.
         if save_to_cache_if_cache_active:
-            self._cache_save(verbose=verbose)
+            self._cache_save()
```

Comment:
Another problem with the current implementation is that the cache will have whatever is requested first, so if a.f90 and b.f90 depend on c.f90 an we do:

psyclone --enable-cache --keep-comments a.f90
psyclone --enable-cache b.f90

b.f90 will have comments from c.f90 imported code, even though we requested not to.

PS: my proposal above of always being true is more consistent because having comments or not will always depend on the "psyclone c.f90" flag, regardless of caching. Not from the a flag from an unrelated file depending if caching is enabled or not.